### PR TITLE
Closes #139 | Implement Dataloader for PhoATIS

### LINKED ISSUE #139: Create dataset loader for PhoATIS
Dataloader name: `phoatis/phoatis.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?phoatis

|     Dataset| phoatis |
|-------------|---|
| Description | This is first public intent detection and slot filling dataset for Vietnamese. The data contains 5871 English utterances from ATIS that are manually translated by professional translators into Vietnamese. |
| Subsets | - |
| Languages | vie |
| Tasks | Slot Filling, Intent Classification |
| License | Unknown (unknown) |
| Homepage | https://github.com/VinAIResearch/JointIDSF |
| HF URL | - |
| Paper URL | https://arxiv.org/abs/2104.02021 |

### PR DESCRIPTION
**Pull request by dehanalkautsar**
Closes #139 .

### Note
- Add some slot labels (i.e. I-flight_number, B-compartment, I-compartment, B-stoploc.airport_code, I-stoploc.airport_code, I-state_name, B-booking_class, I-booking_class, I-aircraft_code) that haven't appeared in the datasets' constant label, but occur in some examples.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by holylovenia** [comment]
@dehanalkautsar Is there any reason for separating the PhoATIS dataloader (#139) into two different ones, i.e., PhoATIS Intent Classification and PhoATIS Slot Filling?

cc: @gentaiscool

**Comment by dehanalkautsar** [comment]
> @dehanalkautsar Is there any reason for separating the PhoATIS dataloader (#139) into two different ones, i.e., PhoATIS Intent Classification and PhoATIS Slot Filling?
> 
> cc: @gentaiscool

@holylovenia I think I misinterpreted the reply you sent in #139 because at that time I hadn't seen the dataloader implementation example that could handle a multi-task dataset. But after this reply (https://github.com/SEACrowd/seacrowd-datahub/issues/139#issuecomment-1875380085) I think I can implement that as a unified dataloader now. I'll update asap when the unified dataloader is ready.

**Comment by dehanalkautsar** [comment]
@holylovenia I have pushed the unified one👍. cc: @gentaiscool

File: seacrowd/sea_datasets/phoatis/phoatis.py
Comment (holylovenia) [inline seacrowd/sea_datasets/phoatis/phoatis.py:154]: `phoatis_intent_cls` --> `phoatis`

File: seacrowd/sea_datasets/phoatis/phoatis.py
Comment (holylovenia) [inline seacrowd/sea_datasets/phoatis/phoatis.py:160]: `phoatis_slot_filling` --> `phoatis`

File: seacrowd/sea_datasets/phoatis/phoatis.py
Comment (holylovenia) [inline seacrowd/sea_datasets/phoatis/phoatis.py:123]: Can we use `"phoatis_intent_cls_seacrowd_text"` instead and adjust the codebase accordingly?

File: seacrowd/sea_datasets/phoatis/phoatis.py
Comment (holylovenia) [inline seacrowd/sea_datasets/phoatis/phoatis.py:130]: Can we use `"phoatis_slot_filling_seacrowd_seq_label"` instead and adjust the codebase accordingly?

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @dehanalkautsar, thanks for your hard work. I found that loading the `seq_label` subset throws this error `ValueError: Invalid string class label I-flight_number`. Could you please check it?

Also, here are the other change requests:

File: seacrowd/sea_datasets/phoatis/phoatis.py
Comment (dehanalkautsar) [reply seacrowd/sea_datasets/phoatis/phoatis.py:130]: Hi @holylovenia sorry for the late reply, actually I have used `"phoatis_intent_cls_seacrowd_text"` earlier. However, an error occurred when I tried to run the test script. Is it okay?

```
ValueError: BuilderConfig 'phoatis_seacrowd_seq_label' not found. Available: ['phoatis_intent_cls_syllable_source', 'phoatis_intent_cls_word_source', 'phoatis_intent_cls_syllable_seacrowd_text', 'phoatis_intent_cls_word_seacrowd_text', 'phoatis_slot_filling_syllable_source', 'phoatis_slot_filling_word_source', 'phoatis_slot_filling_syllable_seacrowd_seq_label', 'phoatis_slot_filling_word_seacrowd_seq_label', 'phoatis_source', 'phoatis_intent_cls_seacrowd_text', 'phoatis_slot_filling_seacrowd_seq_label']
```

File: seacrowd/sea_datasets/phoatis/phoatis.py
Comment (dehanalkautsar) [reply seacrowd/sea_datasets/phoatis/phoatis.py:123]: same comment as the reply below.

File: seacrowd/sea_datasets/phoatis/phoatis.py
Comment (holylovenia) [reply seacrowd/sea_datasets/phoatis/phoatis.py:130]: @dehanalkautsar Did you use a subset id that does not exist, e.g., `phoatis_seacrowd_seq_label`? According to the error message, these are the available subset ids: `Available: ['phoatis_intent_cls_syllable_source', 'phoatis_intent_cls_word_source', 'phoatis_intent_cls_syllable_seacrowd_text', 'phoatis_intent_cls_word_seacrowd_text', 'phoatis_slot_filling_syllable_source', 'phoatis_slot_filling_word_source', 'phoatis_slot_filling_syllable_seacrowd_seq_label', 'phoatis_slot_filling_word_seacrowd_seq_label', 'phoatis_source', 'phoatis_intent_cls_seacrowd_text', 'phoatis_slot_filling_seacrowd_seq_label']`.

Comment (dehanalkautsar) [reply seacrowd/sea_datasets/phoatis/phoatis.py]: Oops, I wasn't aware that the test script has `--schema` and `--subset_id` args. I have revised all of your requested changes @holylovenia. This error `ValueError: Invalid string class label I-flight_number` also has been resolved by adding some labels on `slot_label.txt`.

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @dehanalkautsar, I'm not sure why there are `_syllable` and `_word` sub-subsets for `intent_cls` subsets since it seems they don't have any difference. I also don't think we need multiple `source` schemas; one is enough. What do you think?

**Comment by dehanalkautsar** [comment]
Hi @holylovenia. The reason behind creating two sub-subsets for `intent_cls` was initially to provide all the components in the dataset, because the original dataset also divides them into two sub-subsets for `intent_cls`. But if you want to merge them into just one sub-subset, that is also fine with me. Wdyt?

For the second one, you're right that the source schema needs to be in just one schema.

**Comment by holylovenia** [comment]
> Hi @holylovenia. The reason behind creating two sub-subsets for `intent_cls` was initially to provide all the components in the dataset, because the original dataset also divides them into two sub-subsets for `intent_cls`. But if you want to merge them into just one sub-subset, that is also fine with me. Wdyt?
> 
> For the second one, you're right that the source schema needs to be in just one schema.

I see, thanks for explaining to me. Then we can keep both subsets for `intent_cls` in the `seacrowd` subset and only one `source` schema, @dehanalkautsar.

**Comment by dehanalkautsar** [comment]
Okay @holylovenia, I've removed multiple source schemas, resulting in the final schemas: `['phoatis_intent_cls_syllable_seacrowd_text', 'phoatis_intent_cls_word_seacrowd_text', 'phoatis_slot_filling_syllable_seacrowd_seq_label', 'phoatis_slot_filling_word_seacrowd_seq_label', 'phoatis_source', 'phoatis_intent_cls_seacrowd_text', 'phoatis_slot_filling_seacrowd_seq_label']`. Let me know if you have any suggestions for changes.

**Review by gentaiscool — APPROVED** [review]
lgtm @dehanalkautsar thanks for the contribution. let's wait for @holylovenia for the final review.

**Review by holylovenia — APPROVED** [review]
LGTM!! Thanks a lot for your help, @dehanalkautsar!!

**Pull request merged**
